### DISCUSSION TITLE
I have problems accessing the parameters of the dynamic routes

### DISCUSSION THREAD

**Discussion by Unlucko** [reflex-dev/reflex#3209] [General]
I have a project with dynamic routes which are suposed to be an integer but i don't know how to access the parameter in order to extract the integer information it should have, please help.

**Comment by picklelo** [comment]
@Unlucko can you share the code you're trying?

**Reply by Unlucko** [reply]
It's like this
```
class State(rx.State):
    @rx.var
    def post_id(self) -> int:
         return self.router.page.params.get("pid","...")

@rx.page(route="users/[pid]"):
def post():
    return in_users(State.post_id)
```

It's more like we need the id to make a sqlalchemy query.

**Reply by picklelo** [reply]
What's your `in_users` function?

The following works for me already:

```python
import reflex as rx

class State(rx.State):
    @rx.var
    def post_id(self) -> int:
         return self.router.page.params.get("pid","...")

@rx.page(route="users/[pid]")
def post():
    return rx.fragment(rx.cond(
        State.post_id % 2 == 0,
        "Even",
        "Odd",
    ))

app = rx.App()
```

If your `in_users` is doing a sqlalchemy query - the way to do it would be to create another var that holds the results of that query, and then refer to that var on the frontend. You can't put sqlalchemy (or any random python code) in the frontend, they MUST be in your event handlers.

**Reply by Unlucko** [reply]
```
def in_users(tab: str) -> rx.Component:
    content= []

    try:
        for i in w.get_item_by_id(int(tab)):
            for j in i:
                for k in l.get_item_by_id(j[0]):
                    content.append([j[1],k[1],k[2]])

    except: pass
    try:
        content.append([rx.text_area(tab).value,tab,tab])
    except: pass

    return rx.hstack(
        sidebar2(lessons_sidebar),
        rx.vstack(
            navbar(),
            my_space_tittle(tab),
            my_space(tab, content),
            width="100%"
        )
    )
```
This is the code for in_users, and what we really need is like accessing that number like a python normal number but given what you just said i'm not sure if that's possible

**Reply by picklelo** [reply]
You need to put this within your state / event handler and refer to it there. It won't work in your UI code. See this [diagram](https://github.com/reflex-dev/reflex?tab=readme-ov-file#lets-break-this-down) for the separation of frontend and backend.

**Reply by Unlucko** [reply]
Thanks, this approach was very helpful, and solved the issue.
